packages/remix-react/components.tsx

```diff
@@ -999,20 +1000,31 @@ type HTMLFormSubmitter = HTMLButtonElement | HTMLInputElement;
  * @see https://remix.run/api/remix#useformaction
  */
 export function useFormAction(
-  action = ".",
+  action?: string,
   // TODO: Remove method param in v2 as it's no longer needed and is a breaking change
   method: FormMethod = "get"
 ): string {
   let { id } = useRemixRouteContext();
-  let path = useResolvedPath(action);
-  let search = path.search;
-  let isIndexRoute = id.endsWith("/index");
+  let resolvedPath = useResolvedPath(action ?? ".");
+
+  // Previously we set the default action to ".". The problem with this is that
+  // `useResolvedPath(".")` excludes search params and the hash of the resolved
+  // URL. This is the intended behavior of when "." is specifically provided as
+  // the form action, but inconsistent w/ browsers when the action is omitted.
+  // https://github.com/remix-run/remix/issues/927
+  let location = useLocation();
+  let { search, hash } = resolvedPath;
+  if (action === undefined) {
+    search = location.search;
+    hash = location.hash;
+  }
 
-  if (action === "." && isIndexRoute) {
+  let isIndexRoute = id.endsWith("/index");
+  if ((action === undefined || action === ".") && isIndexRoute) {
```

Comment:
super edge case, but would it make sense to use `(action ?? ".") === "."` here to ensure it's identical logic as the above?  not expected but I think passing `action=null` would result in weird behavior currently since it would match above but not here

Reply:
Kinda hate the way this looks though haha.  Maybe `let resolvedAction = action ?? "."` above?

```suggestion
  if (((action ?? ".") === ".") && isIndexRoute) {
```

Reply:
@brophdawg11 haven't you heard? Everyone uses TS now so that would never happen 😛

Reply:
all your types are belong to us